src/FSharp.Stats/Integration/Differentiation.fs

```diff
@@ -22,3 +26,66 @@ module Differentiation =
         let b  = (x1 - x0 - a*t1*t1)/t1
         (2.*a*t) + b
 
+    ///A two-point estimation is to compute the slope of a nearby secant line through two points.
+    ///This gives an approximations of f'(x) at x respectively to two points "x and x+h"/"x-h and x+h"(depending on the used algorithm) of the function f.
+    ///Choosing a small number h, h represents a small change in x, and it can be either positive or negative.
+    module TwoPointDifferentiation =
+    
+        //correcture of the Array.iterUntili function; here no boundarys are set to the lower and upper end
+        /// Iterates the data array beginning from the startIdx. 
+        /// The step size and direction are implied by magnitude and sign of stepSize. The function returns
+        /// the idx of the first value for which predicate returns true or the end/start of the collection
+        /// is reached (returning None). The predicate function takes the idx of the current value as an additional
+        /// parameter.
+        let private iterUntiliWithoutCutOff (predicate: int -> 'T -> bool) stepSize startIdx (arr: 'T []) =
```

Comment:
```suggestion
        let private iterUntili (predicate: int -> 'T -> bool) stepSize startIdx (arr: 'T []) =
```

Reply:
I wanted to have a different name than "iterUntili" to minimize the possibility for confusion, as the function is not the exact same as the iterUntili function. 
Please let me know if i should still change the function name.